src/app/app.component.ts

```diff
@@ -12,25 +12,27 @@ import { SnipcartEvents } from "@shared/models/types/snipcart-events.type";
 export class AppComponent implements OnInit, OnDestroy {
     private subscriptions: Subscription = new Subscription();
     title = "dak-wholesale";
-    cartEvents!: SnipcartEvents;
+    cartEvents: SnipcartEvents = defaultSnipcartEvents();
```

Comment:
I am not a big fan of this. Colocating types and functions... let's leave as it was previously. Unless you have a specific reason?

Reply:
Originally, in *@shared/models/types/snipcart-events.type.ts* you had
```
export type SnipcartEvents = {
    addingItemSubscription: () => {},
    addedItemSubscription: () => {},
    updatedItemSubscription: () => {},
    removedItemSubscription: () => {},
    orderCompletedSubscription: () => {}
};
```
And you were using it as seen on the diff for this comment chain, which was causing `ERROR TypeError: Cannot set property 'addedItemSubscription' of undefined` since `cartEvents` never had an object assigned to it. Assigning an empty object wouldn't work (`cartEvent: SnipcartEvents = {}`) since the empty object `{}` is missing the defined properties (`addingItemSubscription` etc.), so somewhere in the code there needed to be a default object with all the properties defined to empty functions, hence the change in *@shared/models/types/snipcart-events.type.ts* to:
```
export type SnipcartEvents = {
    snipcartInitializedSubscription: () => void,
    addingItemSubscription: () => void,
    addedItemSubscription: () => void,
    updatedItemSubscription: () => void,
    removedItemSubscription: () => void,
    orderCompletedSubscription: () => void,
};

export function defaultSnipcartEvents(): SnipcartEvents {
    return {
        snipcartInitializedSubscription: () => {},
        addingItemSubscription: () => {},
        addedItemSubscription: () => {},
        updatedItemSubscription: () => {},
        removedItemSubscription: () => {},
        orderCompletedSubscription: () => {},
    };
}
```
I did it this way since I thought it was cleaner than populating *app.component.ts* with the default definition. Is there an alternative that fixes the probably as well? Happy to change to whatever is best.

Reply:
Ahhh I see where the issue is... That was a mistake on my part, when the types were defined as () => {}, it was meant as a function that returns nothing (the curly braces were supposed to identify the function block not the return type).

Complete brain fuck on my part. You are right that we can need to update SnipcartEvents so all functions are void, but there is no need for the defaultSnipcartEvents method since we never need to return empty objects